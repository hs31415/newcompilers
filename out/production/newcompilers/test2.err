Line: 4 Quote didn't match.Line: 9 '&&' didn't match.