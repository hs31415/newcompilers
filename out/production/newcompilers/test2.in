test2=function(double D, integer K, double rc){
     integer a[100];
     double b[43][20];
     char s = "a string 12;
     if(N==1) return(0);
     integer KL=floor(K * rc);         #split N

     if (KL > .5*K && D >= 1)
            KL = 1&;
     else if(KL <= 1 || KL > (K-1))
            KL = ceiling(KL / 2.0)

     D = K - KL

     integer NL = ceiling(N * KL / K)      #split N
     integer NR = N - NL

     return ( 2+(NL * test2(KL, NL, rc) + NR * test2(D, NR, rc )/N))
}
