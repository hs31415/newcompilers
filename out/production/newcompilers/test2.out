<id, test2>
 <=>
 <function>
 <(>
 <double>
<id, D>
 <,>
 <integer>
<id, K>
 <,>
 <double>
<id, rc>
 <)>
 <{>
 <integer>
<id, a>
 <[>
<INT, 100>
 <]>
 <;>
 <double>
<id, b>
 <[>
<INT, 43>
 <]>
 <[>
<INT, 20>
 <]>
 <;>
 <char>
<id, s>
 <=>
Line: 4 Quote didn't match. <if>
 <(>
<id, N>
 <==>
<INT, 1>
 <)>
 <return>
 <(>
<NUM, 0>
 <)>
 <;>
 <integer>
<id, KL>
 <=>
<id, floor>
 <(>
<id, K>
 <*>
<id, rc>
 <)>
 <;>
 <if>
 <(>
<id, KL>
 <>>
<REAL, 0.5>
 <*>
<id, K>
 <&&>
<id, D>
 <>=>
<INT, 1>
 <)>
<id, KL>
 <=>
<INT, 1>
Line: 9 '&&' didn't match. <;>
 <else>
 <if>
 <(>
<id, KL>
 <<=>
<INT, 1>
 <||>
<id, KL>
 <>>
 <(>
<id, K>
 <->
<INT, 1>
 <)>
 <)>
<id, KL>
 <=>
<id, ceiling>
 <(>
<id, KL>
 </>
<REAL, 2.0>
 <)>
<id, D>
 <=>
<id, K>
 <->
<id, KL>
 <integer>
<id, NL>
 <=>
<id, ceiling>
 <(>
<id, N>
 <*>
<id, KL>
 </>
<id, K>
 <)>
 <integer>
<id, NR>
 <=>
<id, N>
 <->
<id, NL>
 <return>
 <(>
<INT, 2>
 <+>
 <(>
<id, NL>
 <*>
<id, test2>
 <(>
<id, KL>
 <,>
<id, NL>
 <,>
<id, rc>
 <)>
 <+>
<id, NR>
 <*>
<id, test2>
 <(>
<id, D>
 <,>
<id, NR>
 <,>
<id, rc>
 <)>
 </>
<id, N>
 <)>
 <)>
 <}>
