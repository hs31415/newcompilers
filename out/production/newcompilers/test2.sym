KEYWORD function
KEYWORD read
KEYWORD write
KEYWORD return
KEYWORD if
KEYWORD else
KEYWORD for
KEYWORD repeat
KEYWORD until
KEYWORD do
KEYWORD while
KEYWORD integer
KEYWORD double
KEYWORD char
IDENTIFIER N
IDENTIFIER floor
IDENTIFIER K
IDENTIFIER rc
IDENTIFIER b
IDENTIFIER NR
IDENTIFIER D
IDENTIFIER KL
IDENTIFIER a
IDENTIFIER test2
IDENTIFIER NL
IDENTIFIER s
IDENTIFIER ceiling
