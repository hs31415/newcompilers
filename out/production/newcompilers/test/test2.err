Line: 1 Invalid token.
Line: 1 Invalid token.
