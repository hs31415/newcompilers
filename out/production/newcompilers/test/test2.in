char c = 'A';
for (integer i = 0; i < 10; i = i + 1) {
    write(c);
}