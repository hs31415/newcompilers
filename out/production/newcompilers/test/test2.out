 <char>
<id, c>
 <=>
 <'A'>
 <;>
 <for>
 <(>
 <integer>
<id, i>
 <=>
<NUM, 0>
 <;>
<id, i>
 <<>
<INT, 10>
 <;>
<id, i>
 <=>
<id, i>
 <+>
<INT, 1>
 <)>
 <{>
 <write>
 <(>
<id, c>
 <)>
 <;>
 <}>
